pkg/blockdeviceclaim/v1alpha1/build.go

```diff
@@ -323,3 +328,28 @@ func (b *Builder) WithBlockVolumeMode(mode corev1.PersistentVolumeMode) *Builder
 
 	return b
 }
+
+// WithBlockDevicePoolName appends (or creates) the BDC Label Selector
+// by setting the provided value to the fixed key
+// openebs.io/block-device-pool
+// This will enable the NDM to pick only devices that
+// match the node (hostname) and block device pool label.
+func (b *Builder) WithBlockDevicePoolName(bdPoolName string) *Builder {
+	if len(bdPoolName) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build BDC object: missing block device pool name"),
+		)
+		return b
+	}
+
+	if b.BDC.Object.Spec.Selector == nil {
```

Comment:
MatchLabels can still be nil even when b.BDC.Object.Spec.Selector is not nil.
If above happens then 353 will panic.

Reply:
good one. Thanks. fixed.